src/Compilers/CSharp/Portable/Symbols/PublicModel/TypeParameterSymbol.cs

```diff
@@ -0,0 +1,109 @@
+﻿// Copyright (c) Microsoft.  All Rights Reserved.  Licensed under the Apache License, Version 2.0.  See License.txt in the project root for license information.
+
+using System.Collections.Immutable;
+using System.Diagnostics;
+
+namespace Microsoft.CodeAnalysis.CSharp.Symbols.PublicModel
+{
+    internal sealed class TypeParameterSymbol : TypeSymbol, ITypeParameterSymbol
+    {
+        private readonly Symbols.TypeParameterSymbol _underlying;
+
+        public TypeParameterSymbol(Symbols.TypeParameterSymbol underlying, CodeAnalysis.NullableAnnotation nullableAnnotation)
+            : base(nullableAnnotation)
+        {
+            Debug.Assert(underlying is object);
+            _underlying = underlying;
+        }
+
+        protected override ITypeSymbol WithNullableAnnotation(CodeAnalysis.NullableAnnotation nullableAnnotation)
+        {
+            Debug.Assert(nullableAnnotation != _underlying.DefaultNullableAnnotation);
+            Debug.Assert(nullableAnnotation != this.NullableAnnotation);
+            return new TypeParameterSymbol(_underlying, nullableAnnotation);
+        }
+
+        internal override Symbols.TypeSymbol UnderlyingTypeSymbol => _underlying;
+        internal override CSharp.Symbol UnderlyingSymbol => _underlying;
+        internal override Symbols.NamespaceOrTypeSymbol UnderlyingNamespaceOrTypeSymbol => _underlying;
+        internal Symbols.TypeParameterSymbol UnderlyingTypeParameterSymbol => _underlying;
+
+#pragma warning disable IDE0055 // Fix formatting. This formatting is correct, need 16.1 for the updated formatter to not flag
```

Comment:
Any chance that this pragma can be removed?

Reply:
> Any chance that this pragma can be removed?

It was present in the original code. Experimenting whether it can be removed is outside of scope of this PR.

---
In reply to: [338830538](https://github.com/dotnet/roslyn/pull/39498#discussion_r338830538) [](ancestors = 338830538)